syntax = "proto3";

import "src/common/base/proto/status.proto";
import "src/carnot/proto/plan.proto";

package pl.carnot.compiler.compilerpb;

option go_package = "compilerpb";

message LineColError {
  uint64 line = 1;
  uint64 column = 2;
  string message = 3;
}
message CompilerError {
  oneof error { LineColError line_col_error = 1; }
}

message CompilerResult {
  pl.statuspb.Status status = 1;
  pl.carnot.carnotpb.Plan logical_plan = 2;
}

message CompilerErrorGroup { repeated CompilerError errors = 1; }
